Sources/AppHealthChecks/ConsulHealthChecks/ConsulHealthChecks.swift

```diff
@@ -0,0 +1,81 @@
+// FS App Health Checks
+// Copyright (C) 2024  FREEDOM SPACE, LLC
+
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Affero General Public License as published
+//  by the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU Affero General Public License for more details.
+//
+//  You should have received a copy of the GNU Affero General Public License
+//  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+//
+//  ConsulHealthChecks.swift
+//
+//
+//  Created by Mykola Buhaiov on 07.02.2024.
+//
+
+import Vapor
+
+/// Service that provides consul health check functionality
+public struct ConsulHealthChecks: ConsulHealthChecksProtocol {
+    /// Instance of app as `Application`
+    public let app: Application
+
+    /// Get response time from consul
+    /// - Returns: `HealthCheckItem`
+    public func getResponseTime() async -> HealthCheckItem {
+        let url = app.consulConfigData?.url ?? Constants.consulUrl
+        let path = app.consulConfigData?.statusPath ?? Constants.consulStatusPath
+        let dateNow = Date().timeIntervalSinceReferenceDate
+        let status = await getStatus()
+        let result = HealthCheckItem(
+            componentId: app.consulConfigData?.id,
+            componentType: .component,
+            observedValue: Date().timeIntervalSinceReferenceDate - dateNow,
+            observedUnit: "s",
+            status: status == .ok ? .pass : .fail,
+            time: app.dateTimeISOFormat.string(from: Date()),
+            output: status != .ok ? "\(url + path)" : nil,
```

Comment:
Need error description. something like this - ERROR: No connection at this url: \() or No connection at this url: \()